rusoto/core/src/xmlerror.rs

```diff
@@ -14,37 +14,38 @@ impl XmlErrorDeserializer {
     pub fn deserialize<T: Peek + Next>(tag_name: &str,
                                        stack: &mut T)
                                        -> Result<XmlError, XmlParseError> {
-        try!(start_element(tag_name, stack));
+        start_element(tag_name, stack)?;
 
         let mut obj = XmlError::default();
 
         loop {
-            match &try!(peek_at_name(stack))[..] {
+            match &peek_at_name(stack)?[..] {
                 "Type" => {
-                    obj.error_type = try!(string_field("Type", stack));
-                    continue;
+                    obj.error_type = string_field("Type", stack)?;
                 }
                 "Code" => {
-                    obj.code = try!(string_field("Code", stack));
-                    continue;
+                    obj.code = string_field("Code", stack)?;
                 }
                 "Message" => {
-                    obj.message = try!(string_field("Message", stack));
-                    continue;
+                    obj.message = string_field("Message", stack)?;
                 }
                 "Detail" => {
-                    try!(start_element("Detail", stack));
+                    start_element("Detail", stack)?;
                     if let Ok(characters) = characters(stack) {
                         obj.detail = Some(characters.to_string());
-                        try!(end_element("Detail", stack));
+                        end_element("Detail", stack)?;
                     }
-                    continue;
+                },
+                "" => {
+                    break
+                },
+                _ => {
```

Comment:
Perhaps it'd be useful to log to debug/trace, or some output that we encountered an unknown tag to help debugging?

Reply:
Agreed, I'll add it.